### PR TITLE
add EventLogFilterParams

### PR DESCRIPTION
**Pull request by ywy2090**

### CODE REVIEW

File: libevent/EventLogFilterParams.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        return (_index < eth::MAXINUM_TOPIC_EVENT_LOG);
+    }
+
+private:
+    int m_groupID;
+    std::string m_filterID;
+    eth::BlockNumber m_startBlock;
+    eth::BlockNumber m_endBlock;
+    AddressHash m_addresses;
+    std::array<h256Hash, 4> m_topics;
```
Comment (wheatli) [inline libevent/EventLogFilterParams.h]: use MAXINUM_TOPIC_EVENT_LOG instead of 4

File: libevent/EventLogFilterParams.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        if (_index < eth::MAXINUM_TOPIC_EVENT_LOG)
+        {
+            m_topics[_index].insert(_t);
+        }
+
+        return (_index < eth::MAXINUM_TOPIC_EVENT_LOG);
+    }
+
+private:
+    int m_groupID;
```
Comment (wheatli) [inline libevent/EventLogFilterParams.h]: use GROUP_ID instead of int

File: libevent/EventLogFilterParams.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
```
Comment (wheatli) [inline libevent/EventLogFilterParams.h]: 2016-2020?

File: libevent/Common.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http:www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
```
Comment (wheatli) [inline libevent/Common.h]: 2018->2020?

File: libethcore/Common.h [hunk 5]
```diff
@@ -72,8 +73,13 @@ static const u256 shannon = exp10<9>();
 static const u256 wei = exp10<0>();
 
 using Nonce = h64;
-using BlockNumber = int64_t;
 using NonceKeyType = u256;
+using BlockNumber = int64_t;
+// the max blocknumber value
+static const BlockNumber MAX_BLOCK_NUMBER = INT64_MAX;
+// the max number of topic in event logd
+static const uint32_t MAXINUM_TOPIC_EVENT_LOG = 4;
```
Comment (JimmyShi22) [inline libethcore/Common.h]: What's about ?
```suggestion
static const uint32_t MAX_NUM_TOPIC_EVENT_LOG = 4;
```

File: libevent/Common.h [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @file Common.h
+ * @author: octopuswang
+ * @date 2019-08-13
+ */
+
+#pragma once
+
+#include <json/json.h>
```
Comment (bxq2011hust) [inline libevent/Common.h]: It seems json is not used here?

File: libevent/EventLogFilterParams.cpp [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+            {  // filterID field not exist
+                strDesc = "filterID field not exit.";
+                break;
+            }
+            filterID = root["filterID"].asString();
+
+            // groupID field
+            if (!root.isMember("groupID"))
+            {  // groupID field not exist
+                strDesc = "groupID field not exit.";
```
Comment (bxq2011hust) [inline libevent/EventLogFilterParams.cpp]: make all this validate operation a function?

File: libevent/EventLogFilterParams.cpp [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+            }
+
+            if (startBlock > endBlock)
+            {  // invalid block range
+                strDesc = "endBlock bigger than startBlock.";
+                break;
+            }
+
+            // address field
+            if (!root.isMember("address"))
```
Comment (bxq2011hust) [inline libevent/EventLogFilterParams.cpp]: make all this validate operation a function?

File: libevent/EventLogFilterParams.cpp [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+            {
+                strDesc = "address field not exit.";
+                break;
+            }
+
+            params =
+                std::make_shared<EventLogFilterParams>(groupID, filterID, startBlock, endBlock);
+            if (root["address"].isArray())
```
Comment (bxq2011hust) [inline libevent/EventLogFilterParams.cpp]: make parse operations a function?

File: libevent/Common.h (hunk 4 above)
Comment (ywy2090) [reply libevent/Common.h]: > 2018->2020?

2019 - 2021

**Pull request closed without merging**